### DISCUSSION TITLE
Failed Uploading workload

### DISCUSSION THREAD

**Discussion by NaveenkumarMD** [microsoft/Microsoft-Fabric-workload-development-sample#158] [Q&A]
<img width="413" alt="Screenshot 2024-11-22 at 5 27 53 PM" src="https://github.com/user-attachments/assets/fbe7010f-b798-4de0-b6d8-efd32461395e">
I have generated my nupkg manifest file and when I try to upload it in the add workload section, it says Failed Uploading workload.
Where can we see the failure logs?

**Comment by ghost** [comment]
Right now they are not showing us any error messages in the UI. Inspect the network using browser developer tools and check the response of the network call for the below endpoint.

`Request URL:
https://wabi-us-central-a-primary-redirect.analysis.windows.net/metadata/extensions/upload
Request Method:
POST`

**Comment by NaveenkumarMD** [comment]
```
{
    "code": "ExtensionsInvalidPackage",
    "pbi.error": {
        "code": "ExtensionsInvalidPackage",
        "details": [
            {
                "code": "Reason",
                "detail": {
                    "type": 1,
                    "value": "InvalidParameter"
                }
            },
            {
                "code": "Target",
                "detail": {
                    "type": 1,
                    "value": "ResourceId"
                }
            },
            {
                "code": "Message",
                "detail": {
                    "type": 1,
                    "value": "Uri is not valid for Production mode - must use HTTPS"
                }
            }
        ],
        "exceptionCulprit": 1
    }
}
```

```
        <AADApp>
          <AppId>dcd3f7b3-9466-45c6-9149-*****</AppId> <!-- Your application ID -->
          <RedirectUri>https://****/60006</RedirectUri>
          <ResourceId>api://localdevinstance/************</ResourceId>
        </AADApp>
```
@vinayslumel  Do we need to change the resourceID to something?

**Comment by NaveenkumarMD** [comment]
One more question.
Also, do we need to complete the registration form given  [form](https://learn.microsoft.com/en-us/fabric/workload-development-kit/publish-workload-flow#process) before uploading the workload?
![image](https://github.com/user-attachments/assets/bcd63829-44ec-4404-b77d-5a3c80f386c5)

**Reply by ghost** [reply]
https://learn.microsoft.com/en-us/fabric/workload-development-kit/workload-cloud-setup

You will have to follow this documentation to publish your workload to your own tenant. And to publish it to the workload hub, you will have to follow the steps in the documentation you shared.

**Reply by vadimazarov** [reply]
> https://learn.microsoft.com/en-us/fabric/workload-development-kit/workload-cloud-setup
> 
> You will have to follow this documentation to publish your workload to your own tenant. And to publish it to the workload hub, you will have to follow the steps in the documentation you shared.

Hi @NaveenkumarMD, indeed the configuration for a package that you can *Upload* differs from the one in Development Mode.
The workload-cloud-setup link contains the required changes

**Comment by NaveenkumarMD** [comment]
The doc doesn't contain about making the workload available to all users in the organization. Shouldn't I upload the workload nupkg anywhere?

**Reply by vadimazarov** [reply]
The workload's availability depends on how you've uploaded/registered your workload with Fabric.
For 'Dev' mode - you are the only one who'll be able to view and use it.
After you've uploaded the nupkg as described in the documentation above - you have an option to **add** the workload to your entire tenant (via the Admin Portal) - as described here - https://learn.microsoft.com/en-us/fabric/workload-development-kit/manage-workload
This makes it available for all users in your tenant.
To make the workload available elsewhere require **publishing** the workload, which entails a certification process described here - https://learn.microsoft.com/en-us/fabric/workload-development-kit/publish-workload-requirements
